internal/app/machined/pkg/controllers/k8s/control_plane_static_pod.go

```diff
@@ -462,8 +462,13 @@ func (ctrl *ControlPlaneStaticPodController) manageAPIServer(ctx context.Context
 					constants.AnnotationStaticPodConfigVersion:     configResource.Metadata().Version().String(),
 				},
 				Labels: map[string]string{
-					"tier":    "control-plane",
-					"k8s-app": k8s.APIServerID,
+					"tier":                         "control-plane",
+					"k8s-app":                      k8s.APIServerID,
+					"app.kubernetes.io/name":       k8s.APIServerID,
+					"app.kubernetes.io/version":    compatibility.VersionFromImageRef(cfg.Image).String(),
+					"app.kubernetes.io/component":  "control-plane",
+					"app.kubernetes.io/part-of":    "Kubernetes",
```

Comment:
I wonder if it should be lower-case?

Reply:
(I don't have kubeadm cluster to check this out, but probably should be same)

Reply:
kubeadm does something different: https://github.com/kubernetes/kubernetes/blob/2fda3e4a650d8e729c509609aa865f0b6d843b4e/cmd/kubeadm/app/util/staticpod/utils.go#L75-L76

Reply:
I'm totally fine for Talos to have more labels, but I guess it makes sense to be somewhat compatible

Reply:
I did some small changes, thank you for your PR